src/commonControls/numberControl/NumberControlAPL.ts

```diff
@@ -0,0 +1,167 @@
+        requestValue: {
+            document: numberPadDocumentGenerator(),
+            dataSource: numberPadDataSourceGenerator(),
+            customHandlingFuncs: [],
+        },
+        requestChangedValue: {
+            document: numberPadDocumentGenerator(),
+            dataSource: numberPadDataSourceGenerator(),
+            customHandlingFuncs: [],
+        },
+    };
+
+    /**
+     * The APL dataSource to use when requesting a number value
+     *
+     */
+    export function numberPadDataSourceGenerator() {
+        return (act: RequestValueAct | RequestChangedValueAct) => {
+            return {
+                numPadData: {
+                    controlId: act.control.id,
+                    headerTitle: i18next.t('NUMBER_CONTROL_DEFAULT_APL_HEADER_TITLE'),
+                    isValidValue: (act.control as NumberControl).state.isValidValue,
+                    validationFailedMessage: (act.control as NumberControl).getAplValidationFailedMessage(),
+                },
+            };
+        };
+    }
+
+    /**
+     * The APL document to use when requesting a number value.
+     * For information on EditText, see:
+     * https://developer.amazon.com/en-US/docs/alexa/alexa-presentation-language/apl-edittext.html
+     *
+     */
+    export function numberPadDocumentGenerator() {
+        return {
+            type: 'APL',
+            version: '1.4',
+            import: [
+                {
+                    name: 'alexa-layouts',
+                    version: '1.2.0',
+                },
+            ],
+            styles: {
+                EditStyle: {
+                    values: [
+                        {
+                            borderWidth: 2,
+                            borderStrokeWidth: 1,
+                            borderColor: 'darkgrey',
+                            hintColor: 'grey',
+                            hintWeight: '200',
+                            fontSize: '20dp',
+                        },
+                        {
+                            when: '${state.focused}',
+                            borderColor: 'green',
+                            borderStrokeWidth: 2,
+                        },
+                    ],
+                },
+            },
+            mainTemplate: {
+                parameters: ['numPadData'],
+                items: {
+                    type: 'Container',
+                    width: '100vw',
+                    height: '100vh',
+                    direction: 'column',
+                    items: [
+                        {
+                            type: 'AlexaBackground',
+                        },
+                        {
+                            type: 'AlexaHeader',
+                            headerTitle: '${numPadData.headerTitle}',
+                            headerDivider: true,
+                            theme: 'dark',
```

Comment:
I don't fully understand the theme system, particularly whether the user gets to set a theme on their device or not.  we should check with UX.  But I have been trying to avoid setting the theme explicitly on the assumption the user might get to choose.
Something to look into.

Reply:
It seems 'theme' is set on a per-component basis by the APL author and isn't device settings-dependent. Also looks like the default across the board is 'dark'. In any case, with what we're currently using in all controls (the default AlexaBackground), 'light' doesn't work well as it assumes a light background and makes the text dark.